pkg/networkservice/chains/nsmgr/server.go

```diff
@@ -68,20 +69,57 @@ type nsmgrServer struct {
 	registry.Registry
 }
 
+type serverOptions struct {
+	authorizeServer networkservice.NetworkServiceServer
+	dialOptions     []grpc.DialOption
+	registryCC      grpc.ClientConnInterface
+}
+
+// Option modifies server option value
+type Option func(o *serverOptions)
+
+// WithDialOptions sets gRPC Dial Options for the server
+func WithDialOptions(dialOptions ...grpc.DialOption) Option {
+	return func(o *serverOptions) {
+		o.dialOptions = dialOptions
+	}
+}
+
+// WithAuthorizeServer sets authorization server chain element
+func WithAuthorizeServer(authorizeServer networkservice.NetworkServiceServer) Option {
+	if authorizeServer == nil {
+		panic("Authorize server cannot be nil")
+	}
+	return func(o *serverOptions) {
+		o.authorizeServer = authorizeServer
+	}
+}
+
+// WithRegistryCC sets client connection to reach the upstream registry, could be nil, in this case only in memory storage will be used.
+func WithRegistryCC(registryCC grpc.ClientConnInterface) Option {
+	return func(o *serverOptions) {
+		o.registryCC = registryCC
+	}
+}
+
 var _ Nsmgr = (*nsmgrServer)(nil)
 
 // NewServer - Creates a new Nsmgr
 //           nsmRegistration - Nsmgr registration
-//           authzServer - authorization server chain element
 //           tokenGenerator - authorization token generator
-//           registryCC - client connection to reach the upstream registry, could be nil, in this case only in memory storage will be used.
-// 			 clientDialOptions -  a grpc.DialOption's to be passed to GRPC connections.
-func NewServer(ctx context.Context, nsmRegistration *registryapi.NetworkServiceEndpoint, authzServer networkservice.NetworkServiceServer, tokenGenerator token.GeneratorFunc, registryCC grpc.ClientConnInterface, clientDialOptions ...grpc.DialOption) Nsmgr {
+func NewServer(ctx context.Context, nsmRegistration *registryapi.NetworkServiceEndpoint, tokenGenerator token.GeneratorFunc, options ...Option) Nsmgr {
```

Comment:
I think with this refactoring we can get rid of using `nsmRegistration *registryapi.NetworkServiceEndpoint`. Let's pass listening url and name via options. 
Note: Name can have a value by default, URL is required parameter.